catalog/core/catalog-core-federationstrategy/src/main/java/ddf/catalog/federation/impl/SortedFederationStrategy.java

```diff
@@ -165,6 +163,7 @@ public void run() {
               site.getId(),
               e);
           processingDetails.add(new ProcessingDetailsImpl(site.getId(), e));
+          Thread.currentThread().interrupt();
```

Comment:
❗️ Not a good idea here since you are in a loop and on the next iteration, you will get another interrupted exception. Instead, you need to set a boolean flag to true as you were interrupted and put a try/finally around the for{} where in the finally, you check if your flag is true and if it is, you then propagate the interruption,

Reply:
@paouelle will have this change out in a sec. Look back at this in an hour or so. Thanks!